File: packages/ui-compose/ui/src/main/kotlin/com/kinetixui/ui/Label.kt
package com.kinetixui.ui

import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.dimensionResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp

/**
 * KinetixLabel — mirrors `packages/ui/src/components/label.tsx`
 * (`text-sm font-medium leading-none`). "leading-none" means line-height
 * equals font-size exactly (a CSS ratio-of-1 concept, not one of the shared
 * type-scale dimens) — expressed here by setting `lineHeight` to the same
 * value as `fontSize`.
 *
 * `color` defaults to the theme's `foreground` but is overridable —
 * [KinetixFieldLabel] uses this to switch to `destructive` when its
 * enclosing [KinetixField] is invalid, mirroring the React source's
 * `invalid && "text-destructive"`.
 */
@Composable
fun KinetixLabel(
    text: String,
    modifier: Modifier = Modifier,
    color: Color = KinetixColorScheme.current.foreground,
) {
    val fontSize = dimensionResource(R.dimen.font_size_body_md).value.sp
    Text(
        text = text,
        modifier = modifier,
        color = color,
        fontWeight = FontWeight.Medium,
        fontSize = fontSize,
        lineHeight = fontSize,
    )
}

File: packages/ui-swiftui/Sources/KinetixUI/Label.swift
//
// Label.swift — KinetixLabel.
//
// Mirrors packages/ui/src/components/label.tsx (`text-sm font-medium
// leading-none`). `color` defaults to the theme `foreground` but is
// overridable — a field wrapper switches it to `destructive` on an
// invalid control, mirroring the React `invalid && "text-destructive"`.
//

import SwiftUI

public struct KinetixLabel: View {
    @Environment(\.kinetixColors) private var colors

    private let text: String
    private let color: Color?

    public init(_ text: String, color: Color? = nil) {
        self.text = text
        self.color = color
    }

    public var body: some View {
        Text(text)
            .font(.system(size: 14, weight: .medium))
            .foregroundStyle(color ?? colors.foreground)
    }
}
